Block/Analytics.php

```diff
@@ -14,15 +14,19 @@ public function __construct(
         \Magento\Framework\App\Action\Context $actionContext,
         \Metrilo\Analytics\Helper\Data        $helper,
         \Magento\Framework\Registry           $registry,
+        \Magento\Framework\View\Page\Title    $pageTitle,
+        \Magento\Framework\UrlInterface       $urlInterface,
```

Comment:
What happened to the idea that the logic for each event should be contained in the class for the particular event?

Reply:
This is result of using a wrong pattern for instantiating the class. Will fix it using the proxy pattern.